skrub/tests/test_sklearn.py

```diff
@@ -1,33 +1,86 @@
-from sklearn.utils.estimator_checks import check_estimator
+import numpy as np
+import pytest
 
-from skrub import (  # TableVectorizer,
+import sklearn
+from sklearn.metrics.pairwise import linear_kernel, pairwise_distances
+from sklearn.utils.estimator_checks import (
+    _is_pairwise_metric,
+    parametrize_with_checks,
+)
+from sklearn.utils._tags import _safe_tags
+
+from skrub import (
     DatetimeEncoder,
+    # FeatureAugmenter,
     GapEncoder,
     MinHashEncoder,
     SimilarityEncoder,
-    TargetEncoder,
+    TableVectorizer,
+    # TargetEncoder,
 )
+from skrub._utils import parse_version
 
 
-def test_sklearn_compatible_DatetimeEncoder():
-    check_estimator(DatetimeEncoder())
-
+def _enforce_estimator_tags_X_monkey_patch(estimator, X, kernel=linear_kernel):
+    """Monkey patch scikit-learn function to create a specific case where to enforce
+    having only strings with some encoders.
+    """
+    # Estimators with `1darray` in `X_types` tag only accept
+    # X of shape (`n_samples`,)
+    if "1darray" in _safe_tags(estimator, key="X_types"):
+        X = X[:, 0]
+    # Estimators with a `requires_positive_X` tag only accept
+    # strictly positive data
+    if _safe_tags(estimator, key="requires_positive_X"):
+        X = X - X.min()
+    if "categorical" in _safe_tags(estimator, key="X_types"):
+        X = np.round((X - X.min()))
+        if "string" in _safe_tags(estimator, key="X_types"):
+            # Note: this part is the monkey patch
+            X = X.astype(object)
+            for i in range(X.shape[0]):
+                for j in range(X.shape[1]):
+                    X[i, j] = str(X[i, j])
+        elif _safe_tags(estimator, key="allow_nan"):
+            X = X.astype(np.float64)
```

Comment:
Why float64 and not 32?

Reply:
The scikit-learn conversion will be `np.float64` so it will be better for all common test.

Reply:
:+1: